tools/clang/lib/SPIRV/InitListHandler.cpp

```diff
@@ -275,19 +269,19 @@ InitListHandler::createInitForVectorType(QualType elemType, uint32_t count,
     while (tryToSplitStruct() || tryToSplitConstantArray())
       ;
 
-    const Expr *init = initializers.back();
+    auto *init = initializers.back();
+    const auto initType = init->getAstResultType();
 
-    if (hlsl::IsHLSLVecType(init->getType()) &&
-        hlsl::GetHLSLVecSize(init->getType()) == count) {
+    uint32_t elemCount = 0;
+    if (isVectorType(initType, nullptr, &elemCount) && elemCount == count) {
```

Comment:
So, in the past we had:
```
if (hlsl::IsHLSLVecType(init->getType()) &&  hlsl::GetHLSLVecSize(init->getType()) == count) {
```
If `count` is 1, this condition would be **true** for `int1` type.

but now we have:
```
if (isVectorType(initType, nullptr, &elemCount) && elemCount == count) {
```
If `count` is 1, this condition would be **false** for `int1` type.